### ISSUE TITLE
Add support for converting MAP and ARRAY types from Velox to Substrait and back

### ISSUE THREAD

**Issue by mbasmanova** [facebookincubator/velox#2270]
See velox/substrait/tests/VeloxToSubstraitTypeTest.cpp

```
  // Array type is not supported yet.
  ASSERT_ANY_THROW(testTypeConversion(ARRAY(BIGINT())));

  // Map type is not supported yet.
  ASSERT_ANY_THROW(testTypeConversion(MAP(BIGINT(), DOUBLE())));
```

**Comment by mbasmanova** [comment]
CC: @zhejiangxiaomai

**Comment by mbasmanova** [comment]
Fixed in #2393 and #2392
